### PR TITLE
feat(lsp): more consistent ReactDTS filtering

### PR DESCRIPTION
**Pull request by H4ckint0sh**
Don't handle jump to definition manually and take different scenarios as typescript versions in consideration.

### CODE REVIEW

File: lua/lsp/servers/tsserver.lua [hunk 1]
```diff
@@ -56,44 +56,42 @@ local on_attach = function(client, bufnr)
...
+
+  return filtered
+end
+
+local function filterReactDTS(value)
+  -- Depending on typescript version either uri or targrtUri is returned
+  if value.uri then
+    return string.match(value.uri, '%.d.ts') == nil
+  elseif value.targetUri
+    return string.match(value.targrtUri, '%.d.ts') == nil
```
Comment (ecosse3) [inline lua/lsp/servers/tsserver.lua:79]: `end` is missing

image

**Review by ecosse3 — CHANGES_REQUESTED** [review]

File: lua/lsp/servers/tsserver.lua (hunk 1 above)
Comment (ecosse3) [reply lua/lsp/servers/tsserver.lua:79]: + `then` is missing in line 78

Comment (H4ckint0sh) [reply lua/lsp/servers/tsserver.lua:79]: Oh, good catch, sorry i missed those.

**Comment by H4ckint0sh** [comment]
> I understand it works the same way I did but it has cleaner code, so it's basically refactored?

I would not call it a refactor as you handling the jump manually.

**Review by ecosse3 — APPROVED** [review]
Thanks for contribution!

**Comment by H4ckint0sh** [comment]
> @H4ckint0sh Could you please change target branch to `dev` instead of `masters?
And how do i do that? should I make a new PR?

**Pull request merged**